shared/src/main/scala/scala/xml/Utility.scala

```diff
@@ -243,6 +243,12 @@ object Utility extends AnyRef with parsing.TokenTests {
       }
     }
 
+  // Checks if node when converted to string is an entity ref
+  def checkNodeForEntityRef(n: Node): Boolean = {
+    val st = n.toString
+    st(0) == '&' && st(st.length - 1) == ';'
```

Comment:
Ok, bummer.  Thanks for looking in to that.  I really appreciate it.

I guess I'll need to come up with some more test cases to better understand how the `ConstructingParser` parses entities.